api/envoy/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto

```diff
@@ -0,0 +1,416 @@
+  //         '@type': type.googleapis.com/envoy.extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings
+  //         no_assignment_behavior:
+  //           fallback_rate_limit:
+  //             blanket_rule: DENY_ALL
+  //
+  // .. attention::
+  //  The first matched group wins. Once the request is matched into a bucket, matcher
+  //  evaluation ends.
+  //
+  // Use ``on_no_match`` field to assign the catch-all bucket. If a request is not matched
+  // into any bucket, and there's no  ``on_no_match`` field configured, the request will be
+  // ALLOWED by default. It will NOT be reported to the RLQS server.
+  //
+  // Refer to :ref:`Unified Matcher API <envoy_v3_api_msg_.xds.type.matcher.v3.Matcher>`
+  // documentation for more information on the matcher trees.
+  xds.type.matcher.v3.Matcher bucket_matchers = 3 [(validate.rules).message = {required: true}];
+}
+
+// Per-route and per-host configuration overrides. The more specific definition completely
+// overrides the less specific definition.
+message RateLimitQuotaOverride {
+  // The application domain to use when calling the service. This enables sharing the quota
+  // server between different applications without fear of overlap.
+  // E.g., "envoy".
+  //
+  // If empty, inherits the value from the less specific definition.
+  string domain = 1;
+
+  // The match tree to use for grouping incoming requests into buckets.
+  //
+  // If set, fully overrides the bucket matchers provided on the less specific definition.
+  // If not set, inherits the value from the less specific definition.
+  //
+  // See usage example: :ref:`RateLimitQuotaFilterConfig.bucket_matchers
+  // <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaFilterConfig.bucket_matchers>`.
+  xds.type.matcher.v3.Matcher bucket_matchers = 2;
+}
+
+// Rate Limit Quota Bucket Settings to apply on the successful ``bucket_matchers`` match.
+//
+// Specify this message in the :ref:`Matcher.OnMatch.action
+// <envoy_v3_api_field_.xds.type.matcher.v3.Matcher.OnMatch.action>` field of the
+// ``bucket_matchers`` matcher tree to assign the matched requests to the Quota Bucket.
+// Usage example: :ref:`RateLimitQuotaFilterConfig.bucket_matchers
+// <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaFilterConfig.bucket_matchers>`.
+// [#next-free-field: 6]
+message RateLimitQuotaBucketSettings {
+  // Configures the behavior after the first request has been matched to the bucket, and before the
+  // the RLQS server returns the first quota assignment.
+  message NoAssignmentBehavior {
+    // Blocking behavior: wait for the RLQS server to return the first assignment before
+    // processing any requests.
+    // [#not-implemented-hide:]
+    message BlockingBehavior {
+      // Limit the time the requests are blocked until the RLQS server to returns
+      // the first assignment. If the server doesn't respond within the configured time,
```

Comment:
What circuit breakers are in play here? This could easily lead to overload without protection? Do this need to be independently configured given that this is pre-cluster circuit breaker?

Reply:
This option will not be used in Envoy, as there are other timeouts that control that. If this value is not needed for gRPC, I think we should just remove it.

Reply:
+1 if this is in the config we need circuit breakers, so let's just remove for now and we can add later?

Reply:
Removed in https://github.com/envoyproxy/envoy/pull/19793/commits/f7665e145b75fb886c124ed6beaa1df35cd9fd8a.